src/main/java/com/hms_networks/americas/sc/extensions/util/RawNumberValueUtils.java

```diff
@@ -0,0 +1,313 @@
+   * @since 1.15.7
+   */
+  public static Object getValueMax(Object value1, Object value2) {
+    // Compare values by type
+    Object maxValue;
+    if (value1 instanceof Integer && value2 instanceof Integer) {
+      // Compare values
+      Integer int1 = (Integer) value1;
+      Integer int2 = (Integer) value2;
+      if (int1.intValue() > int2.intValue()) {
+        maxValue = int1;
+      } else {
+        maxValue = int2;
+      }
+    } else if (value1 instanceof Long && value2 instanceof Long) {
+      // Compare values
+      Long long1 = (Long) value1;
+      Long long2 = (Long) value2;
+      if (long1.longValue() > long2.longValue()) {
+        maxValue = long1;
+      } else {
+        maxValue = long2;
+      }
+    } else if (value1 instanceof Float && value2 instanceof Float) {
+      // Compare values
+      Float float1 = (Float) value1;
+      Float float2 = (Float) value2;
+      if (float1.floatValue() > float2.floatValue()) {
+        maxValue = float1;
+      } else {
+        maxValue = float2;
+      }
+    } else if (value1 instanceof Double && value2 instanceof Double) {
+      // Compare values
+      Double double1 = (Double) value1;
+      Double double2 = (Double) value2;
+      if (double1.doubleValue() > double2.doubleValue()) {
+        maxValue = double1;
+      } else {
+        maxValue = double2;
+      }
+    } else if (value1 instanceof Short && value2 instanceof Short) {
+      // Compare values
+      Short short1 = (Short) value1;
+      Short short2 = (Short) value2;
+      if (short1.shortValue() > short2.shortValue()) {
+        maxValue = short1;
+      } else {
+        maxValue = short2;
+      }
+    } else if (value1 instanceof Byte && value2 instanceof Byte) {
+      // Compare values
+      Byte byte1 = (Byte) value1;
+      Byte byte2 = (Byte) value2;
+      if (byte1.byteValue() > byte2.byteValue()) {
+        maxValue = byte1;
+      } else {
+        maxValue = byte2;
+      }
+    } else {
+      // Throw exception
+      throw new IllegalArgumentException(
+          "Cannot compare values of type "
+              + value1.getClass().getName()
+              + " and "
+              + value2.getClass().getName());
+    }
+
+    // Return max value
+    return maxValue;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueAvg(Object[] values) {
+    // Get sum of values
+    Object sum = getValueSum(values);
+
+    // Get average of values
+    Object avg = getValueDivide(sum, values.length);
+
+    // Return average
+    return avg;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
```

Comment:
I don't think there is any checking if values are the same type.  
If values are not types that support addition? 
Should add comment that caller is responsible values being a homogenous list

Reply:
https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/pull/81#discussion_r1385311719

`Values` does not necessarily need to be a list of all the same types, as long as they are all implementations of `Number`. Checking has been added/ensured for this.